pkg/image/qualify/image.go

```diff
@@ -0,0 +1,42 @@
+package qualify
+
+import (
+	"strings"
+
+	"k8s.io/kubernetes/pkg/util/parsers"
+)
+
+// SplitImageName splits a docker image reference into the domain
+// component and everything after the domain. An empty string is
+// returned if there is no domain component. This function will first
+// validate that image is a valid reference, returning an error if it
+// is not.
+//
+// Examples inputs and results for the domain component:
+//
+//   "busybox"                      -> domain is ""
+//   "foo/busybox"                  -> domain is ""
+//   "localhost/foo/busybox"        -> domain is "localhost"
+//   "localhost:5000/foo/busybox"   -> domain is "localhost:5000"
+//   "gcr.io/busybox"               -> domain is "gcr.io"
+//   "gcr.io/foo/busybox"           -> domain is "gcr.io"
+//   "docker.io/busybox"            -> domain is "docker.io"
+//   "docker.io/library/busybox"    -> domain is "docker.io"
+//   "library/busybox:v1"           -> domain is ""
+func SplitImageName(image string) (string, string, error) {
```

Comment:
We already have such parsers. There's also docker's [reference package](https://github.com/docker/distribution/blob/master/reference/reference.go#L179) which as de facto referential implementation.